### ISSUE TITLE
Windows generating FMU2.0:  libgcc_s_seh-1.dll can not be resolved

### ISSUE THREAD

**Issue by Zbm1996** [OpenModelica/OpenModelica#10031]
While compiling OMC on Windows through CMake. After I add  
-DOM_ENABLE_GUI_CLIENTS=OFF
to disable gui function on MSYS cmaking process, I couldn't generate fmu successfully but everything was right on Linux. 

My CMake command is showed below:
cmake -S . -B build_cmake -Wno-dev -G "MSYS Makefiles" -DCMAKE_BUILD_TYPE=Release -DOM_USE_CCACHE=OFF -DOM_ENABLE_GUI_CLIENTS=OFF
The cmake process ran successfully, but when I used omc.exe to generate fmu from the Van.mo mentioned above. I received the error below:
[linux_error.txt](https://github.com/OpenModelica/OpenModelica/files/10379667/linux_error.txt)

![image](https://user-images.githubusercontent.com/25106682/211480536-dc50a18a-d5a8-4d32-9647-fffdba18fa09.png)
This was similar to the error in which I couldn't resolve api-ms-win-core-apiquery-l1-1-0.dll. However, I ldd the related dll and the result was showed below.
![image](https://user-images.githubusercontent.com/25106682/211481010-c267945f-f307-4e52-aad5-c88b3e8186a2.png)
Actually,  libgcc_s_seh-1.dll could be found and accessed through the given path "/mingw64/bin/libgcc_s_seh-1.dll" in MinGW shell.

**Comment by Zbm1996** [comment]
> Can you add the `Van.fmutmp/sources/CMakeList.txt` here? I'm not sure what line 71 is. Also there should be some log file, I think in `Van.fmutmp/resources`. Or remove the build directory and run CMake again and log the output yourself.
> 
> When I execute
> 
> ```modelica
> loadFile("Van.mo"); getErrorString();
> cd("temp"); getErrorString();
> setCommandLineOptions("--fmiFlags=s:euler"); getErrorString();
> setCommandLineOptions("--fmuCMakeBuild=true --fmuRuntimeDepends=modelica"); getErrorString();
> buildModelFMU(Van, version="2.0",  fmuType="cs", platforms={"dynamic"}); getErrorString();
> ```
> 
> Van.mos
> 
> ```
> model Van
>   Real y_1 (start=2);
>   Real y_2 (start=0);
>   parameter Real eta=1000;
>  
> equation
>   der(y_1)= y_2; 
>   der(y_2)=-y_1 + (1 - y_1*y_1)*y_2*eta;
> end Van;
> ```
> 
> my executable doesn't have MINGW libs linked:
> 
> ```shell
> # ldd Van.dll
>         ntdll.dll => /c/WINDOWS/SYSTEM32/ntdll.dll (0x7ffb9cd30000)
>         KERNEL32.DLL => /c/WINDOWS/System32/KERNEL32.DLL (0x7ffb9c110000)
>         KERNELBASE.dll => /c/WINDOWS/System32/KERNELBASE.dll (0x7ffb9a780000)
>         apphelp.dll => /c/WINDOWS/SYSTEM32/apphelp.dll (0x7ffb971b0000)
>         msvcrt.dll => /c/WINDOWS/System32/msvcrt.dll (0x7ffb9c010000)
>         libwinpthread-1.dll => /d/workspace/Testitesttest/issue-10031/temp/FMU/binaries/win64/libwinpthread-1.dll (0x7ffb8fdb0000)
> ```

The related CMakeLists.txt and the CMakeLists.txt.in files are showed below:
[CMakeLists.txt](https://github.com/OpenModelica/OpenModelica/files/10382545/CMakeLists.txt)
[CMakeLists.txt.in.txt](https://github.com/OpenModelica/OpenModelica/files/10382552/CMakeLists.txt.in.txt)
I think the line 71 error message related cmake_install.cmake file is this one.
![image](https://user-images.githubusercontent.com/25106682/211547785-be2b46f7-f8e7-444f-ac07-e1091573aea7.png)
The file is also attached here. 
[cmake_install.cmake.txt](https://github.com/OpenModelica/OpenModelica/files/10382567/cmake_install.cmake.txt)
Thanks for help!

**Comment by mahge** [comment]
Something that might be relevant is that **we do not install** (as in OpenModelica CMake build install) the necessary MSYS/MinGW DLLs to the installation bin directory **if we are not building OMEdit**. This means `libgcc_s_seh-1.dll` will NOT be installed into the installation bin directory if the user is building without GUI support, i.e., has specified `-DOM_ENABLE_GUI_CLIENTS=OFF`.

The reasoning behind this was the assumption that omc will always be run (when directly called) from an MSYS/MinGW shell. So there was no need to copy the DLLs used by omc since they will be on the PATH added by MSYS/MinGW shells themselves.
On the other hand OMEdit and other GUI Clients are usually used from Windows (Windows Explorer) which means we need to make the necessary DLLs available somehow and copying them to the installation bin folder (where OMEdit is located) was the straightforward and appropriate solution.

That said I am still not sure why your FMU depends on `libgcc_s_seh-1.dll` (I am not saying it should not but @AnHeuermann reports it does not for him). If it should depend on `libgcc_s_seh-1.dll` the we might have an explanation for the error because the CMakeLists.txt you have attached above specifies 
```cmake
...
# Line 23
set(RUNTIME_DEPENDENCIES_LEVEL modelica)
...
# Line 171
set(EXTERNAL_LIBDIRECTORIES "D:/OpenModelica/build_cmake/install_cmake/bin")

# Line 232
...
elseif(RUNTIME_DEPENDENCIES_LEVEL STREQUAL "modelica")

  install(TARGETS ${FMU_NAME}

    RUNTIME_DEPENDENCIES

      DIRECTORIES ${EXTERNAL_LIBDIRECTORIES} # <----- This execludes MSYS/MInGW directories from the search path.

      PRE_EXCLUDE_REGEXES "api-ms-" "ext-ms-"

      POST_EXCLUDE_REGEXES "^\\/lib.*" "^\\/usr\\/lib.*" "^\\/usr\\/local\\/lib.*" ".*system32/.*\\.dll"

    PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE

    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}

    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}

    RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

Which means it will **look for dependency DLLs ONLY in** `EXTERNAL_LIBDIRECTORIES="D:/OpenModelica/build_cmake/install_cmake/bin"` (@AnHeuermann this is how it works right?). However, `libgcc_s_seh-1.dll` is not installed to that place.

----------------------------------------------------------

We can install the dependency DLLs for omc the same way we do for OMEdit to cover cases like this. @AnHeuermann what do you think? Perhaps this can be part of #9834 as well.

**Comment by AnHeuermann** [comment]
> Which means it will look for dependency DLLs ONLY in EXTERNAL_LIBDIRECTORIES="D:/OpenModelica/build_cmake/install_cmake/bin" (@AnHeuermann this is how it works right?). 

Yes, on Windows systems CMake only searches inside `DIRECTORIES`.

> That said I am still not sure why your FMU depends on libgcc_s_seh-1.dll (I am not saying it should not but @AnHeuermann reports it does not for him).

It could be that in my case a static version of libgcc is compiled into the executable. Using `nm Van.dll` I can see that some functions like `__gcc_register_frame` are in my library.

**Comment by Zbm1996** [comment]
> > Which means it will look for dependency DLLs ONLY in EXTERNAL_LIBDIRECTORIES="D:/OpenModelica/build_cmake/install_cmake/bin" (@AnHeuermann this is how it works right?).
> 
> Yes, on Windows systems CMake only searches inside `DIRECTORIES`.
> 
> > That said I am still not sure why your FMU depends on libgcc_s_seh-1.dll (I am not saying it should not but @AnHeuermann reports it does not for him).
> 
> It could be that in my case a static version of libgcc is compiled into the executable. Using `nm Van.dll` I can see that some functions like `__gcc_register_frame` are in my library.

So, you are working on this bug report right? Do I need to provide some extra information?

**Comment by AnHeuermann** [comment]
I'm currently doing other stuff.
Did you try building OpenModelica GUI clients (removing `-DOM_ENABLE_GUI_CLIENTS=OFF`)? If that is a solution for you I would ignore it for now.

Maybe we'll implement @mahge's approach in that case:

> We can install the dependency DLLs for omc the same way we do for OMEdit to cover cases like this. @AnHeuermann what do you think? Perhaps this can be part of https://github.com/OpenModelica/OpenModelica/pull/9834 as well.
